BeVolt/BSP/Simulator/DataGeneration/PLL.py

```diff
@@ -0,0 +1,22 @@
+import csv
+
+file = "BSP/Simulator/DataGeneration/Data/PLL.csv"
+
+"""
+@brief change the clock frequency
+    Function to be called by simulate.py
+"""
+CurrentFrequency = 16000000
+
+
+def Change_Frequency(newFrequency):
+    with open(file, 'w') as csvfile:
```

Comment:
Add this line `os.makedirs(os.path.dirname(file), exist_ok=True)` before opening the file. This checks to see if the Data directory exists and if it doesn't, then it'll create it to place the PLL.csv file.

Reply:
Also add locks whenever accessing the csv file. Under issues, there is a ticket called _Simulator: Locking files during read and writes_ which shows what to add to lock the code